test/wallet/electrum.cpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2011-2017 libbitcoin developers (see AUTHORS)
+ *
+ * This file is part of libbitcoin.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <boost/test/unit_test.hpp>
+#include <bitcoin/bitcoin.hpp>
+
+using namespace bc;
+using namespace bc::wallet;
+
+BOOST_AUTO_TEST_SUITE(electrum_tests)
+
+// These test cases were generated using electrum code and verified to
+// match output from:
+//
+// https://github.com/spesmilo/electrum/blob/master/lib/mnemonic.py
+
+BOOST_AUTO_TEST_CASE(electrum__decode_mnemonic__no_passphrase_1)
+{
+    static const word_list mnemonic{ { "foo", "bar", "baz"} };
+    const auto seed = electrum::decode_mnemonic(mnemonic);
+    BOOST_REQUIRE_EQUAL(encode_base16(seed), "c4033901dd61ba26cfd0a1cf1ceb4b347606635aa3cb951eb6e819d58beedc04dd400a2e600d783c83c75879d6538abeecc7bb1b292b2a4d775d348d5d686427");
+}
+
+BOOST_AUTO_TEST_CASE(electrum__decode_mnemonic__no_passphrase_2)
+{
+    static const word_list mnemonic{ { "giggle", "crush", "argue", "inflict",
+        "wear", "defy", "combine", "evolve", "tiger", "spatial", "crumble", "fury" } };
+    const auto seed = electrum::decode_mnemonic(mnemonic);
```

Comment:
I was probably ambiguous in the last style recommendation on this point. The "wrap" allowed in this case is for balanced bracing... basically for more readable declaration of complex structures, maybe not so helpful here:
```cpp
static const word_list mnemonic
{
    {
        "giggle", "crush", "argue", "inflict", "wear", "defy", "combine", "evolve", "tiger", "spatial", "crumble", "fury"
    }
};
```